app/controllers/api/v1/rents_controller.rb

```diff
@@ -0,0 +1,23 @@
+module Api
+  module V1
+    class RentsController < ApplicationController
+      before_action :authenticate_api_v1_user!, only: %i[index create]
+      def index
+        render_paginated Rent.filter(params.slice(:user_id, :book_id))
```

Comment:
In Rails, when indexing and serializing models which have nested associations, you need to explicitly tell this query to "include" them to avoid making a lot of unnecessary queries.
You can read this https://apidock.com/rails/ActiveRecord/QueryMethods/includes
and this http://tomdallimore.com/blog/includes-vs-joins-in-rails-when-and-where/

Ask me directly if this is not clear enough :)